packages/api/src/hooks/useCurrencyConfig.ts

```diff
@@ -0,0 +1,88 @@
+            const crypto_config = crypto_config_data?.crypto_config?.currencies_config[currency];
+
+            return {
+                ...currency_config,
+                ...crypto_config,
+                /** determine if the currency is a `crypto` currency */
+                is_crypto: currency_config?.type === 'crypto',
+                /** determine if the currency is a `fiat` currency */
+                is_fiat: currency_config?.type === 'fiat',
+                /** determine if the currency is `Australian Dollar` */
+                is_AUD: currency === 'AUD',
+                /** determine if the currency is `US Dollar` */
+                is_USD: currency === 'USD',
+                /** determine if the currency is `Euro` */
+                is_EUR: currency === 'EUR',
+                /** determine if the currency is `Pound Sterling` */
+                is_GBP: currency === 'GBP',
+                /** determine if the currency is `Bitcoin` */
+                is_BTC: currency === 'BTC',
+                /** determine if the currency is `Ethereum` */
+                is_ETH: currency === 'ETH',
+                /** determine if the currency is `Litecoin` */
+                is_LTC: currency === 'LTC',
+                /** determine if the currency is `Multi-Collateral DAI` */
+                is_DAI: currency === 'DAI',
+                /** determine if the currency is `IDK` */
+                is_IDK: currency === 'IDK',
+                /** determine if the currency is `Paxos Standard` */
+                is_PAX: currency === 'PAX',
+                /** determine if the currency is `Binary Coin` */
+                is_USB: currency === 'USB',
+                /** determine if the currency is `Tether Omni` */
+                is_USDT: currency === 'UST',
+                /** determine if the currency is `True USD` */
+                is_TUSD: currency === 'TUSD',
+                /** determine if the currency is `Binance USD` */
+                is_BUSD: currency === 'BUSD',
+                /** determine if the currency is `STATIS Euro` */
+                is_EURS: currency === 'EURS',
+                /** determine if the currency is `Tether ERC20` */
+                is_eUSDT: currency === 'eUSDT',
+                /** determine if the currency is `Tether TRC20` */
+                is_tUSDT: currency === 'tUSDT',
+                /** determine if the currency is `USD Coin` */
+                is_USDC: currency === 'USDC',
+                /** determine if the currency is `USDK` */
+                is_USDK: currency === 'USDK',
+                /** Currency code */
+                code: currency,
+                /** Currency display code */
+                display_code: currency === 'UST' ? 'USDT' : currency,
+                /** Local asset name for the currency icon. ex: `IcCurrencyUsd` for `USD` */
+                icon: `IcCurrency${currency[0].toUpperCase() + currency.slice(1).toLowerCase()}`,
```

Comment:
@thisyahlen-deriv do we want to keep the icons in the data layer?

Reply:
Nope, will remove it since we have different currency name convention for wallets. Will do this in component level instead